src/chord/private/chord/Logger.cpp

```diff
@@ -0,0 +1,402 @@
+
+// C++ Header File(s)
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+
+// Code Specific Header Files(s)
+#include "chord/chord.h"
+#include "chord/Logger.h"
+
+using namespace std;
+using namespace Chord;
+
+Logger* Logger::m_Instance = 0;
+std::ofstream Logger::m_File;
+
+// Log file name. File name should be change from here only
+//const string logFileName = "MyLogFile.log";
+std::string logFileName = "";
+
+Logger::Logger()
+{
+   //m_File.open(logFileName.c_str(), ios::out|ios::app);
+   readConfig();
+   m_LogLevel	= LOG_LEVEL_TRACE;
+   m_LogType	= FILE_LOG;
+
+   // Initialize mutex
+#ifdef WIN32
+   InitializeCriticalSection(&m_Mutex);
+#else
+   int ret=0;
+   ret = pthread_mutexattr_settype(&m_Attr, PTHREAD_MUTEX_ERRORCHECK_NP);
+   if(ret != 0)
+   {   
+      printf("Logger::Logger() -- Mutex attribute not initialize!!\n");
+      exit(0);
+   }   
+   ret = pthread_mutex_init(&m_Mutex,&m_Attr);
+   if(ret != 0)
+   {   
+      printf("Logger::Logger() -- Mutex not initialize!!\n");
+      exit(0);
+   }   
+#endif
+}
+
+Logger::~Logger()
+{
+   m_File.close();
+#ifdef WIN32
+   DeleteCriticalSection(&m_Mutex);
+#else
+   pthread_mutexattr_destroy(&m_Attr);
+   pthread_mutex_destroy(&m_Mutex);
+#endif
+}
+
+Logger* Logger::getInstance() throw ()
+{
+    if (m_Instance == 0)
+    {
+        m_Instance = new Logger();
+        if (logFileName.size() > 0)
+        {
+            m_File.open(logFileName.c_str(), ios::out|ios::app);
+        }
+        else
+        {
+            printf("File not found\n");
+        }
+    }
+    return m_Instance;
+}
+
+void Logger::lock()
+{
+#ifdef WIN32
+   EnterCriticalSection(&m_Mutex);
+#else
+   pthread_mutex_lock(&m_Mutex);
+#endif
+}
+
+void Logger::unlock()
+{
+#ifdef WIN32
+   LeaveCriticalSection(&m_Mutex);
+#else
+   pthread_mutex_unlock(&m_Mutex);
+#endif
+}
+
+void Logger::logIntoFile(std::string& data)
+{
+   lock();
+   m_File << getCurrentTime() << "  " << data << endl;
+   unlock();
+}
+
+void Logger::logOnConsole(std::string& data)
+{
+   cout << getCurrentTime() << "  " << data << endl;
+}
+
+string Logger::getCurrentTime()
+{
+   string currTime;
+   //Current date/time based on current time
+   time_t now = time(0); 
+   // Convert current time to string
+   currTime.assign(ctime(&now));
+
+   // Last charactor of currentTime is "\n", so remove it
+   string currentTime = currTime.substr(0, currTime.size()-1);
+   return currentTime;
+}
+
+void Logger :: readConfig()
+{
+	try
+	{
+		std::ifstream infile("/home/sagar55/Downloads/chord/bin/LogConfig");
+		std::getline(infile, logFileName);
+		infile.close();
+	}
+	catch(const ifstream::failure& e)
+	{
+		cout << "File does not exist" << endl;
+	}
+}
+
+void Logger :: chord_print(LogLevel level, const std::string& text, LogType type) throw()
+{

```

Comment:
1. better LogLevel  can be const reference
2. Are we using LogType type, anywhere in this function?